src/Microsoft.AspNetCore.SignalR.Core/HubConnectionHandler.cs

```diff
@@ -184,7 +184,7 @@ public class HubConnectionHandler<THub> : ConnectionHandler where THub : Hub
                             {
                                 // Don't wait on the result of execution, continue processing other
                                 // incoming messages on this connection.
-                                _ = _dispatcher.DispatchMessageAsync(connection, message);
+                                await _dispatcher.DispatchMessageAsync(connection, message);
```

Comment:
Does this mean only one message can be invoked at a time for a connection? What happens if a hub method takes 5 seconds to complete because it is calling a slow DB? Do all other messages wait for it to finish?

Reply:
Scenario: I am connected to jabbr Core. I send David a private message containing a link to github. On the server jabbr parses the message and sees the github link, makes a request to github to include more details. At the same time I am sending messages in another chat room. Will those other messages only show up once the private message has finished being processed?

Reply:
It does, but I believe this is also how ASP.NET SignalR works. This helps propagate back-pressure on the server back up through the client. If the server is backed up waiting to send data (in response to received messages), it slows down processing received messages, which eventually slows down the client, etc...

If you want to invoke a long-running thing in the background, you can fire and forget a task in the body of your Hub method and use `IHubContext<T>` to send a response to the Client via a Client invocation when the long-running task is complete.

Reply:
Ok, as long as everyone who knows more about SignalR understands and expects that behavior. It doesn't seem like something can can easily be undone. If V1 runs messages serially then V2 won't be able to change to run in parallel without breaking some hearts.

I think it should be added as a comment here, and also spelled out in documentation. Devs are use to firing 10 requests at MVC and having them run in parallel.

Reply:
Although I guess if there is load balancing and multiple instances of the SignalR server then messages could still end up running in parallel so people shouldn't rely on that logic.

Reply:
Yes this is how old SignalR worked when you used websockets. Regardless this is a better way to handle dispatching. Running multiple calls to the hub in parallel means there's no self throttling.

Reply:
ok!